pkg/server/server_test.go

```diff
@@ -1373,13 +1423,58 @@ type repo
 }
 
 func TestDefaultMaxConcurrentReadSettings(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+
 	cfg := serverconfig.DefaultConfig()
 	require.EqualValues(t, math.MaxUint32, cfg.MaxConcurrentReadsForCheck)
 	require.EqualValues(t, math.MaxUint32, cfg.MaxConcurrentReadsForListObjects)
 
 	s := MustNewServerWithOpts(
 		WithDatastore(memory.New()),
 	)
+	t.Cleanup(s.Close)
 	require.EqualValues(t, math.MaxUint32, s.maxConcurrentReadsForCheck)
 	require.EqualValues(t, math.MaxUint32, s.maxConcurrentReadsForListObjects)
 }
+
+func TestRateLimitedCheckResolver(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	t.Run("default_rate_limited_check_resolver_disabled", func(t *testing.T) {
+		cfg := serverconfig.DefaultConfig()
+		require.False(t, cfg.DispatchThrottling.Enabled)
+
+		ds := memory.New()
+		t.Cleanup(ds.Close)
+		s := MustNewServerWithOpts(
+			WithDatastore(ds),
+		)
+		t.Cleanup(s.Close)
+		require.Nil(t, s.rateLimitedCheckResolver)
+		require.False(t, s.rateLimitedCheckResolverEnabled)
+	})
+
+	t.Run("enable_rate_limited_check_resolver_enabled", func(t *testing.T) {
+		ds := memory.New()
+		t.Cleanup(ds.Close)
+		s := MustNewServerWithOpts(
+			WithDatastore(ds),
+			WithRateLimitedCheckResolverEnabled(true),
+			WithRateLimitedCheckResolverMediumPriorityShaper(20),
+			WithRateLimitedCheckResolverMediumPriorityLevel(30),
+			WithRateLimitedCheckResolverLowPriorityShaper(40),
+			WithRateLimitedCheckResolverLowPriorityLevel(50),
+		)
+		t.Cleanup(s.Close)
+
+		require.NotNil(t, s.rateLimitedCheckResolver)
+		require.True(t, s.rateLimitedCheckResolverEnabled)
```

Comment:
Please also add an assertion that asserts the server's CheckResolver is of type `CycleDetectionCheckResolver`.

```go
_, ok := s.checkResolver.(*graph.CycleDetectionCheckResolver)
require.True(t, ok)
```

Reply:
done